### PR TITLE
prov/rxd: cq, cntr, flag fixes

### PR DESCRIPTION
**Pull request by aingerson**
- Add util_ep->msg_flags
- Pre-calculate flags
- Correctly use cntr functions
- Use util_cq write functions

- Also includes small fabtests fix related to completions

### CODE REVIEW

File: prov/rxd/src/rxd_cq.c [hunk 1]
```diff
@@ -137,8 +133,17 @@ static void rxd_complete_rx(struct rxd_ep *ep, struct rxd_x_entry *rx_entry)
...
+		goto out;
 	}
 
+	if (rx_entry->cq_entry.flags & FI_REMOTE_CQ_DATA ||
+	    (!(rx_entry->flags & RXD_NO_RX_COMP) &&
+	      rx_entry->cq_entry.flags & FI_RECV))
+		rx_cq->write_fn(rx_cq, &rx_entry->cq_entry);
+
+	assert(rx_entry->op < ofi_op_last);
+	ofi_ep_rx_cntr_inc_funcs[rx_entry->op](&ep->util_ep);
```
Comment (shefty) [inline prov/rxd/src/rxd_cq.c]: Hmm... moving this assertion into a common location would be best.  We could define:
```
static inline ofi_ep_rx_cntr_inc(struct util_ep *ep, u8 opcode)
{
    assert(opcode < ofi_op_last)
    ofi_ep_rx_cntr_inc_funcs[opcode](ep);
}
```
add only access the function array through that.

Comment (shefty) [reply prov/rxd/src/rxd_cq.c]: btw - this could be added as a separate change

**Comment by aingerson** [comment]
Updated to move the assert into common code. @a-ilango there is now a small rxm change. Would you take a look?

File: include/ofi_util.h [hunk 2]
```diff
@@ -329,6 +329,18 @@ typedef void (*ofi_ep_cntr_inc_func)(struct util_ep *);
...
 
+static inline void ofi_ep_tx_cntr_inc_func(struct util_ep *ep, uint8_t op)
+{
+	assert(op < ofi_op_last);
+	ofi_ep_tx_cntr_inc_funcs[op](ep);
+}
+
+static inline void ofi_ep_rx_cntr_inc_func(struct util_ep *ep, uint8_t op)
+{
+	assert(op < ofi_op_last);
```
Comment (a-ilango) [inline include/ofi_util.h]: Should this be renamed to ofi_op_max? ofi_op_last seems to imply last (previous) op.

Comment (shefty) [reply include/ofi_util.h]: I agree max is more common for enum ranges.  This could go in a separate PR

**Pull request merged**